--- src/alphafold3/model/model_config.py.ori
+++ src/alphafold3/model/model_config.py
@@ -16,6 +16,16 @@ from typing import Literal, TypeAlias
 from alphafold3.common import base_config
 from alphafold3.jax.attention import attention
 
+gpu_run = False
+gpu_total_mem = None
+try:
+  import nvidia_smi
+  nvidia_smi.nvmlInit()
+  gpu_0 = nvidia_smi.nvmlDeviceGetHandleByIndex(0)
+  gpu_run = True
+  gpu_total_mem = nvidia_smi.nvmlDeviceGetMemoryInfo(gpu_0).total / 1024 **3
+except nvidia_smi.NVMLError as msg:
+  pass
 
 _Shape2DType: TypeAlias = tuple[int | None, int | None]
 
@@ -24,9 +34,23 @@ class GlobalConfig(base_config.BaseConfig):
   bfloat16: Literal['all', 'none', 'intermediate'] = 'all'
   final_init: Literal['zeros', 'linear'] = 'zeros'
   pair_attention_chunk_size: Sequence[_Shape2DType] = ((1536, 128), (None, 32))
+  #default values
   pair_transition_shard_spec: Sequence[_Shape2DType] = (
       (2048, None),
       (None, 1024),
   )
+  if gpu_run and gpu_total_mem:
+    if gpu_total_mem >= 80.0:
+      pair_transition_shard_spec: Sequence[_Shape2DType] = (
+        (2048, None),
+        (None, 1024),
+      )
+    else:
+      pair_transition_shard_spec: Sequence[_Shape2DType] = (
+        (2048, None),
+        (3072, 1024),
+        (None, 512),
+      )
+
   # Note: flash_attention_implementation = 'xla' means no flash attention.
   flash_attention_implementation: attention.Implementation = 'triton'
